app/src/main/java/de/westnordost/streetcomplete/quests/smoothness/AddSmoothnessForm.kt

```diff
@@ -0,0 +1,166 @@
+            in ALL_PATHS_EXCEPT_STEPS -> R.string.quest_smoothness_description_very_horrible
+            else -> R.string.quest_smoothness_description_very_horrible
+        }
+        Smoothness.IMPASSABLE -> when (highwayTag) {
+            in ALL_PATHS_EXCEPT_STEPS -> R.string.quest_smoothness_description_impassable
+            else -> R.string.quest_smoothness_description_impassable
+        }
+    }
+
+    private fun getImage(smoothness: Smoothness) = when (smoothness) {
+        Smoothness.EXCELLENT -> when (surfaceTag) {
+            "paving_stones" -> R.drawable.surface_paving_stones_excellent
+            else -> R.drawable.surface_asphalt_excellent
+        }
+        Smoothness.GOOD -> when (surfaceTag) {
+            "paving_stones" -> R.drawable.surface_paving_stones_good
+            else -> R.drawable.surface_asphalt_good
+        }
+        Smoothness.INTERMEDIATE -> when (surfaceTag) {
+            "paving_stones" -> R.drawable.surface_paving_stones_intermediate
+            "sett" -> R.drawable.surface_sett_intermediate
+            "compacted" -> R.drawable.surface_compacted_intermediate
+            else -> R.drawable.surface_asphalt_intermediate
+        }
+        Smoothness.BAD -> when (surfaceTag) {
+            "paving_stones" -> R.drawable.surface_paving_stones_bad
+            "sett" -> R.drawable.surface_sett_bad
+            "gravel" -> R.drawable.surface_gravel_bad
+            "compacted" -> R.drawable.surface_compacted_bad
+            else -> R.drawable.surface_asphalt_bad
+        }
+        Smoothness.VERY_BAD -> when (surfaceTag) {
+            "paving_stones" -> R.drawable.surface_paving_stones_very_bad
+            "sett" -> R.drawable.surface_sett_very_bad
+            else -> R.drawable.surface_asphalt_very_bad
+        }
+        // TODO: need images for the values below (and replace the ones above)
+        Smoothness.HORRIBLE  -> R.drawable.surface_paved_area
+        Smoothness.VERY_HORRIBLE -> R.drawable.surface_paved_area
+        Smoothness.IMPASSABLE -> R.drawable.surface_paved_area
+    }
+
+    private fun surfaceWrong() {
+        val surfaceType = Surface.values().find { it.osmValue == surfaceTag }
+
+        if (surfaceType != null) {
+            showWrongSurfaceDialog(surfaceType)
+        } else {
+            // this should not happen: smoothness quest is not asked for unknown surfaces
+            composeNote()
+        }
+    }
+
+    private fun showWrongSurfaceDialog(surface: Surface) {
+        val inflater = LayoutInflater.from(requireContext())
+        val inner = inflater.inflate(R.layout.dialog_quest_smoothness_wrong_surface, null, false)
+        ItemViewHolder(inner.findViewById(R.id.item_view)).bind(surface.asItem())
+
+        AlertDialog.Builder(requireContext())
+            .setView(inner)
+            .setPositiveButton(R.string.quest_generic_hasFeature_yes) { _, _ -> composeNote() } // TODO: maybe do something else?
```

Comment:
I find it somewhat unlikely that the user first selects that "surface doesn't match images" but then decides that after all, it is a (e.g.) asphalt surface after all. 
Maybe rather the question is, whether we need this at all instead of a generic "are you sure?" dialog? The only situation I can think of is that the user can't decide because it is either so bad or the damage he sees doesn't match well with any of the shown damages in the images. Wouldn't it make sense for the user to answer "Can't say" in that case?

Reply:
I had thought more about paving stones, which can look very diverse and already the surface .
Then there could be compacted vs gravel, or some weird tracks I found which basically look like old asphalt covered with a thick layer of gravel instead of fixing the surface.

So I thought I might make sense to show the user what is currently tagged before removing the surface.

Reply:
Ok. Then, the "yes" option should be named "Yes (Leave Note)"

Reply:
done